raco/algebra.py

```diff
@@ -421,23 +421,37 @@ def shortStr(self):
         return self.opname()
 
 
-class UnionAll(IdenticalSchemeBinaryOperator):
+class UnionAll(NaryOperator):
 
     """Bag union."""
 
-    def __init__(self, left=None, right=None):
-        BinaryOperator.__init__(self, left, right)
+    def __init__(self, children=None):
+        NaryOperator.__init__(self, children)
+
+    def partitioning(self):
+        return RepresentationProperties()
```

Comment:
Is this overly conservative? If all relations are identically partitioned can we keep the partitioning?